src/app/state/reducers/datavault.ts

```diff
@@ -0,0 +1,38 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+export interface DataVaultKey {
+  key: string,
+  content: string[]
+}
+
+export interface DataVaultState {
+  hasDataVault: boolean
+  data: DataVaultKey[]
+}
+
+export const initialState: DataVaultState = {
+  hasDataVault: false,
```

Comment:
Why do we need this?

Reply:
I debated if this was a good idea, or if it would be better to check context.

It is currently used in the navigation to hide the navigation item, but will be used on the dashboard when we get the max size limit and storage used implemented.

Because it was used in multiple screens, I was thinking redux. Do you think checking the context would be better?

Reply:
Yes. Duplicating instance value for same logic is bad idea. It can derive in inconsistency or “I forgot to update” in the future

Reply:
Okay, I'll fix this first thing tomorrow.

Reply:
It has been removed.